core/src/main/java/org/ehcache/core/Ehcache.java

```diff
@@ -130,7 +137,7 @@ protected void doRemoveAll(final Set<? extends K> keys) throws BulkCacheWritingE
 
   @Override
   public ValueHolder<V> doPutIfAbsent(final K key, final V value, Consumer<Boolean> put) throws StoreAccessException {
-    ValueHolder<V> result = store.putIfAbsent(key, value);
+    ValueHolder<V> result = store.putIfAbsent(key, value, put);
     if(result == null) {
```

Comment:
Whats going on here?  Why are we both passing the consumer in, and then also explicitly calling it based on the method return.  Can we just not pass it in then?

Reply:
The same though cam tom me as well for the first time & I discussed the same with Henri. Now the reason is described here. It was there earlier as well, just that since loader-writer got pushed to Store level, this had to come here as well. https://github.com/ehcache/ehcache3/pull/2467/commits/bb7f52208e3143c9a2c2b70aa12bfa10706ec2e5

Reply:
The same though cam tom me as well for the first time & I discussed the same with Henri. Now the reason is described here. It was there earlier as well, just that since loader-writer got pushed to Store level, this had to come here as well. https://github.com/ehcache/ehcache3/pull/2467/commits/bb7f52208e3143c9a2c2b70aa12bfa10706ec2e5

Reply:
See comment below